apps/api/src/overall/overall.service.ts

```diff
@@ -556,25 +558,35 @@ async function getProjects(
     }
   }
 
-  // UX Tests KPI updates - Latest average success rate for UX tests with Validation only
-  const kpiUxTestsSuccessRates = projectsData
+const calculateAverage = (numbers) => numbers.reduce((acc, num) => acc + num, 0) / numbers.length;
+
+const kpiUxTestsSuccessRates = projectsData
     .flatMap((project) => project.uxTests)
     .filter((test) => test.test_type === 'Validation' && test.success_rate)
     .map((test) => test.success_rate);
 
-  const testsCompleted = kpiUxTestsSuccessRates.length;
+const avgTestSuccessAvg = calculateAverage(kpiUxTestsSuccessRates);
+
+const taskSuccessRatesRecord: Record<string, number[]> = projectsData
+  .flatMap((project) => project.uxTests)
+  .reduce((acc, test) => {
+    if (test.test_type === 'Validation' && test.success_rate && test.task && test.title) {
```

Comment:
Probably should have looked this over before merging, but I think you need to change this to have:
`(test.success_rate || test.success_rate === 0)`

Reply:
Yes, please change post-haste! Thank you! :)